ddtrace/tracer/span.go

```diff
@@ -266,31 +266,6 @@ func (s *Span) SetTag(key string, value interface{}) {
 	s.setMeta(key, fmt.Sprint(value))
 }
 
-// AddLink sets a casuality link to another span via a spanContext. It also
-// stores details about the link in an attributes map.
-func (s *Span) AddLink(spanContext *SpanContext, attributes map[string]string) {
-	traceIDHex := spanContext.TraceID()
-	traceIDHigh, _ := strconv.ParseUint(traceIDHex[:8], 16, 64)
-
-	samplingDecision, hasSamplingDecision := spanContext.SamplingPriority()
-	var flags uint32
-	if hasSamplingDecision && samplingDecision >= ext.PriorityAutoKeep {
-		flags = uint32(1<<31 | 1)
-	} else if hasSamplingDecision {
-		flags = uint32(1 << 31)
-	} else {
-		flags = uint32(0)
-	}
-
-	s.spanLinks = append(s.spanLinks, SpanLink{
-		TraceID:     spanContext.TraceIDLower(),
-		TraceIDHigh: traceIDHigh,
-		SpanID:      spanContext.SpanID(),
-		Attributes:  attributes,
-		Flags:       flags,
-	})
-}
```

Comment:
@rarguelloF Just to double check: we don't longer need this logic, just the ability to append span links, right? The user is responsible for setting Flags and everything.

Reply:
I'm not sure about all this, but the reality is this function is not used internally by our contribs, and the existing function only requires an already crafted `SpanLink`. Maybe we could expose an utility function `NewSpanLink` with this logic? cc @mabdinur